onboard/cv/main.cpp

```diff
@@ -108,30 +112,46 @@ int main() {
     arTags[0].distance = -1;
     arTags[1].distance = -1;
     #if TB_DETECTION
-      pair<Point2f, double> tennisBall = findTennisBall(src, depth_img);
-      if(tennisBall.second >= 0){
-        Point2f center = tennisBall.first;
-        float dist = depth_img.at<float>(center.y, center.x);
-        if (!isnormal(dist) || dist > 5.0) {
-          dist = (TENNIS_BALL_PIXEL_SIZE / tennisBall.second);
+      tagPair = detector.findARTags(src, depth_img);
+
+      //update both tags in LCM message
+      //first tag
+      if(tagPair.first.id == -1){//no tag found
+        if(left_tag_buffer <= 5){//send the buffered tag
```

Comment:
I would do some testing in practice to see if the number 5 is suitable or not given what Caleb said about how often this code gets called (I would intuitively think this number needs to be higher).